### ISSUE TITLE
license

### ISSUE THREAD

**Issue by benkeser** [CoVPN/correlates_reporting#124]
We should probably pick a license to release under

**Comment by nhejazi** [comment]
I'll look into this and add one this week. I think it's worth investing some time in reading exactly what makes sense for public analysis code of this flavor.

**Comment by nhejazi** [comment]
From some review of license types and recommendations, I actually think this is kind of a hard problem for a code base like this one. Usually, I rely on the MIT license, which is _permissive_ and allows derivative work without much more than license preservation (i.e., derivative work can be closed source). Basically, the downside here is that others can build off of this code base, but keep any all additions closed. I've also worked on projects that use more restrictive (_copyleft_) licenses (e.g., GPL3), which require any derivative work to basically also be open source, often with license preservation. A relevant downside in this case is that it may be less tempting to build off of this work, since all derivative work would have to be open source as well. In my review, I found https://choosealicense.com/licenses/ and https://tldrlegal.com/ to be helpful reading.

Since the purpose of this code base is to deliver on the analyses outlined in the publicly available SAP while possibly serving as a point on which further contributions may be built (e.g., a fork using some of this code), my view is that a copyleft license would be most appropriate, since this will ensure that all derivative work will also be required to be open source. For this, I'd recommend GPL3 or AGPL3, neither of which are particularly permissive. Curious for your take on this @benkeser
